### PR TITLE
Add nightly performance tests into Gitlab NV pipeline

### PR DESCRIPTION
**Pull request by ystefinko**
Add small fixes to shell script. 
Add heaptrack and heaptrack report converting to svg, then saving it as GitLab Pages for human checks.

Relates-To: OLPEDGE-746

Signed-off-by: Yaroslav Stefinko <ext-yaroslav.stefinko@here.com>

### CODE REVIEW

File: scripts/linux/nv/gitlab_test_performance.sh [hunk 1]
```diff
@@ -38,11 +38,49 @@ echo ">>> Local Server started for further performance test ... >>>"
...
+else
+    echo "OK. Full list of tests passed. "
+fi
+
+set +e
+ls -la ./*.gz
+
+mkdir reports/heaptrack
+mkdir heaptrack
+git clone --depth=1 https://github.com/brendangregg/FlameGraph # third party dependency needed for pretty graph generation below
```
Comment (diachenko-mischa) [inline scripts/linux/nv/gitlab_test_performance.sh]: Please, move comment above

File: scripts/linux/nv/gitlab_test_performance.sh [hunk 2]
```diff
@@ -38,11 +38,49 @@ echo ">>> Local Server started for further performance test ... >>>"
...
+do
+    heaptrack_print --print-leaks \
+      --print-flamegraph heaptrack/flamegraph_${archive_name}.data \
+      --file ${archive_name}.gz > reports/heaptrack/report_${archive_name}.txt
+    cat heaptrack/report_${archive_name}.txt
+    ./FlameGraph/flamegraph.pl --title="Flame Graph: ${archive_name}" heaptrack/flamegraph_${archive_name}.data > reports/heaptrack/flamegraph_${archive_name}.svg # pretty graph generation
+    cat reports/heaptrack/flamegraph_${archive_name}.svg >> heaptrack_report.html
+done
+cp heaptrack_report.html reports
+#cat heaptrack_report.html
```
Comment (diachenko-mischa) [inline scripts/linux/nv/gitlab_test_performance.sh]: Remove commented code

File: scripts/linux/nv/gitlab_test_performance.sh [hunk 3]
```diff
@@ -38,11 +38,49 @@ echo ">>> Local Server started for further performance test ... >>>"
...
+mkdir heaptrack
+git clone --depth=1 https://github.com/brendangregg/FlameGraph # third party dependency needed for pretty graph generation below
+
+for archive_name in short_test_null_cache short_test_memory_cache short_test_disk_cache
+do
+    heaptrack_print --print-leaks \
+      --print-flamegraph heaptrack/flamegraph_${archive_name}.data \
+      --file ${archive_name}.gz > reports/heaptrack/report_${archive_name}.txt
+    cat heaptrack/report_${archive_name}.txt
+    ./FlameGraph/flamegraph.pl --title="Flame Graph: ${archive_name}" heaptrack/flamegraph_${archive_name}.data > reports/heaptrack/flamegraph_${archive_name}.svg # pretty graph generation
```
Comment (diachenko-mischa) [inline scripts/linux/nv/gitlab_test_performance.sh]: Move comment above

File: scripts/linux/nv/gitlab_test_performance.sh (hunk 1 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_performance.sh]: thanks, done.

File: scripts/linux/nv/gitlab_test_performance.sh (hunk 2 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_performance.sh]: thanks, done.

File: scripts/linux/nv/gitlab_test_performance.sh (hunk 3 above)
Comment (ystefinko) [reply scripts/linux/nv/gitlab_test_performance.sh]: thanks, done.

**Comment by ystefinko** [comment]
> Your commit message contains a typo:
> "Add **heaptack** and heaptrack report converting to svg, then saving it as GitLab Pages for human checks."

thanks, done.

**Pull request merged**